test/e2e/flowaggregator_test.go

```diff
@@ -102,18 +102,33 @@ const (
 )
 
 func TestFlowAggregator(t *testing.T) {
-	skipIfDualStackCluster(t)
-	data, isIPv6, err := setupTestWithIPFIXCollector(t)
+	data, v4Enabled, v6Enabled, err := setupTestWithIPFIXCollector(t)
 	if err != nil {
 		t.Fatalf("Error when setting up test: %v", err)
 	}
 	defer teardownTest(t, data)
 	defer teardownFlowAggregator(t, data)
 
-	podAIP, podBIP, podCIP, svcB, svcC, err := createPerftestPods(data, isIPv6)
+	podAIP, podBIP, podCIP, err := createPerftestPods(data)
 	if err != nil {
-		t.Fatalf("Error when creating perftest pods and services: %v", err)
+		t.Fatalf("Error when creating perftest pods: %v", err)
 	}
+
+	if v4Enabled {
+		t.Run("IPv4", func(t *testing.T) { testHelper(t, data, podAIP, podBIP, podCIP, false) })
+	}
+
+	if v6Enabled {
+		t.Run("IPv6", func(t *testing.T) { testHelper(t, data, podAIP, podBIP, podCIP, true) })
+	}
+}
+
+func testHelper(t *testing.T, data *TestData, podAIP *PodIPs, podBIP *PodIPs, podCIP *PodIPs, isIPv6 bool) {
+	svcB, svcC, err := createPerftestServices(data, isIPv6)
+	if err != nil {
+		t.Fatalf("Error when creating perftest services: %v", err)
+	}
+	defer deletePerftestServices(data)
```

Comment:
don't you want to log the error if there is one?

Reply:
Thanks, addressed.